packages/client/wallets/aa/src/blockchain/wallets/EVMAAWallet.ts

```diff
@@ -1,406 +1,65 @@
-                            ...this.getLegacyTransactionFeesParamsIfApply(),
-                        });
-                        transaction = await publicClient.writeContract(request);
-                        break;
-                    }
-                    case "sft": {
-                        tokenId = evmToken.tokenId;
-                        const { request } = await publicClient.simulateContract({
-                            account: this.account,
-                            address: contractAddress,
-                            abi: erc1155,
-                            functionName: "safeTransferFrom",
-                            args: [this.getAddress(), toAddress, tokenId, (config as SFTTransferType).quantity, "0x00"],
-                            ...this.getLegacyTransactionFeesParamsIfApply(),
-                        });
-                        transaction = await publicClient.writeContract(request);
-                        break;
-                    }
-                    case "nft": {
-                        tokenId = evmToken.tokenId;
-                        const { request } = await publicClient.simulateContract({
-                            account: this.account,
-                            address: contractAddress,
-                            abi: erc721,
-                            functionName: "safeTransferFrom",
-                            args: [this.getAddress(), toAddress, tokenId],
-                            ...this.getLegacyTransactionFeesParamsIfApply(),
-                        });
-                        transaction = await publicClient.writeContract(request);
-                        break;
-                    }
-                    default: {
-                        throw new WalletSdkError(`Token not supported`);
-                    }
-                }
-
-                if (transaction != null) {
-                    return transaction;
-                } else {
-                    throw new TransferError(
-                        `Error transferring token ${evmToken.contractAddress}
-                    ${!tokenId ? "" : ` tokenId=${tokenId}`}
-                    ${!transaction ? "" : ` with transaction hash ${transaction}`}`
-                    );
-                }
-            } catch (error) {
-                logError("[TRANSFER] - ERROR_TRANSFERRING_TOKEN", {
-                    service: SCW_SERVICE,
-                    error: errorToJSON(error),
-                    tokenId: tokenId,
-                    contractAddress: evmToken.contractAddress,
-                    chain: evmToken.chain,
-                });
-                throw new TransferError(
-                    `Error transferring token ${evmToken.contractAddress}${tokenId == null ? "" : `:${tokenId}}`}`
-                );
-            }
-        });
-    }
-
-    getSigner<Type extends SignerType>(type: Type): SignerMap[Type] {
```

Comment:
the assumption we had when startng was that ethers was more likely to be used than viem (at the time true)
It's hard to imagine that won't be a re-emerging req in the future, so I'd make a push to keep the function signature the same–`getSigner("viem")`